core/store/src/main/java/mil/nga/giat/geowave/core/store/util/DataStoreUtils.java

```diff
@@ -552,16 +607,11 @@ else if ((vis2 == null) || (vis2.length == 0)) {
 			final AdapterStore adapterStore,
 			final QueryFilter clientFilter,
 			final PrimaryIndex index,
-			final ScanCallback<T> scanCallback ) {
-		final ByteBuffer idBuf = row.getAdapterAndDataId();
-		final int totalLength = idBuf.array().length;
-		final byte[] dataId = new byte[idBuf.getInt()];
-		idBuf.get(dataId);
-
-		final byte[] adapterId = new byte[totalLength - dataId.length - 4];
-		idBuf.get(adapterId);
+			final ScanCallback<T, R> scanCallback ) {
+		final byte[] dataId = row.getDataId();
+		final byte[] adapterId = row.getAdapterId();
 		final GeowaveRowId rowId = new GeowaveRowId(
```

Comment:
The params for this rowId appear to be in the wrong order. It's passing index, dataId, adapterId, but the constructor is defined as dataId, adapterId, index.